### ISSUE TITLE
Bug: Able to silence `react-hooks/set-state-in-effect` by moving `setState()` into `useEffect()`

### ISSUE THREAD

**Issue by Joe-Lonsdale** [eslint/eslint#20963]
### Environment

Node version: 22.16.0
npm version: 10.9.2
Local ESLint version: 10.4.1
Global ESLint version: 10.4.1
Operating System: Windows 11 Enterprise

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
import { defineConfig, globalIgnores } from "eslint/config";
import nextVitals from "eslint-config-next/core-web-vitals";
import nextTs from "eslint-config-next/typescript";

const eslintConfig = defineConfig([
  ...nextVitals,
  ...nextTs,
  // Override default ignores of eslint-config-next.
  globalIgnores([
    // Default ignores of eslint-config-next:
    ".next/**",
    "out/**",
    "build/**",
    "next-env.d.ts",
  ]),
]);

export default eslintConfig;

```
</details>

An `eslint(react-hooks/set-state-in-effect)` error is raised from the below code.
```js
    const [state, setState] = useState(1);
    const updateState = () => {
        setState(2);
    }
    useEffect(() => {
        updateState();
    },[])
```
But no linting error is raised from the below code
```js
    const [state, setState] = useState(1);
    useEffect(() => {
        const updateState = () => {
            setState(2);
        }
        updateState();
    },[])
```

### What did you expect to happen?

I expect the error to be raised in both cases

### What actually happened?

The error is only raised when the function performing the `setState()` is not inside the `useEffect()`

### Link to Minimal Reproducible Example

n/a

### Participation

- [ ] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by mdjermanovic** [comment]
This is related to [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks), which isn't maintained by us. Please open an issue on the plugin repo:

https://github.com/facebook/react
